jdaviz/core/loaders/resolvers/resolver.py

```diff
@@ -498,6 +615,122 @@ def on_observation_select_changed(self, _=None):
                                                    sender=self, color="error"))
             self.file_table_populated = False
 
+        # Update footprint selection to match table selection
+        if self.parsed_input_is_query and self.treat_table_as_query and self._footprint_groups:
+            selected_indices = set()
+            for row in self.observation_table.selected_rows:
+                idx = self.observation_table.items.index(row)
+                selected_indices.add(idx)
+
+            # Update footprint marks based on selection
+            for idx, marks in self._footprint_groups.items():
+                selected = idx in selected_indices
+                for mark in marks:
+                    mark.selected = selected
+
+    def toggle_custom_toolbar(self):
+        """Override to control footprint display when toolbar is toggled."""
+        if not self.custom_toolbar_enabled:
+            if self.parsed_input_is_query and self.treat_table_as_query:
+                self._display_observation_footprints()
+            super().toggle_custom_toolbar()
+        else:
+            super().toggle_custom_toolbar()
+            if self.parsed_input_is_query and self.treat_table_as_query:
+                self._remove_observation_footprints()
+
+    def _on_region_select(self, msg):
+        """Handle footprint click events."""
+        if not self._footprint_marks:
+            return
+        if not (self.parsed_input_is_query and self.treat_table_as_query):
+            return
+
+        click_x, click_y = msg.x, msg.y
+        selected_idx = find_closest_polygon_mark(click_x, click_y, self._footprint_marks)
+
+        if selected_idx is not None:
+            # Get selected rows from the table
+            currently_selected = set()
+            if hasattr(self, 'observation_table') and self.observation_table is not None:
+                for row in self.observation_table.selected_rows:
+                    idx = self.observation_table.items.index(row)
+                    currently_selected.add(idx)
+
+            # Toggle behavior
+            if selected_idx in currently_selected:
+                currently_selected.discard(selected_idx)
+                for mark in self._footprint_groups.get(selected_idx, []):
+                    mark.selected = False
+            else:
+                currently_selected.add(selected_idx)
+                if selected_idx in self._footprint_groups:
+                    for mark in self._footprint_groups[selected_idx]:
+                        mark.selected = True
```

Comment:
There are already-implemented selection methods on the viewer, see

https://github.com/spacetelescope/jdaviz/blob/964f372be83fa2ac15858cc1c803a074d89d7694/jdaviz/configs/imviz/plugins/viewers.py#L609-L633

Reply:
How about we add a callback to the `RegionOverlay.selected` traitlet that triggers the corresponding table row selection?